### PR TITLE
plugin/dnstap: create ToMessage function

### PR DESCRIPTION
**Pull request by miekg**
Add ToMessage function.

### CODE REVIEW

**Comment by miekg** [comment]
as said in slack a function inside plugin/dnstap might be better, but is more work

File: plugin/dnstap/const.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// Mirror some of the tap. constants so we can reference them from this plugin
+// and not the vendored directory meaning out-of-tree middleware can use this.
+
+import (
+	tap "github.com/dnstap/golang-dnstap"
+)
+
+// Mirror constants from the golang-dnstap package.
+const (
+	SocketProtocolTCP        = tap.SocketProtocol_TCP
```
Comment (2buisson) [inline plugin/dnstap/const.go]: Would be best placed in dnstap/msg because those are values for field Type of payload type tap.Message, only a detail though.

Comment (miekg) [reply plugin/dnstap/const.go]: all of them? sgtm

**Comment by miekg** [comment]
note sure, if we should merge this.

**Comment by johnbelamaric** [comment]
/integration

let's see what happens...

**Comment by miekg** [comment]
@johnbelamaric wow! :)

**Comment by johnbelamaric** [comment]
It's all @chrisohaver not me :)

**Comment by miekg** [comment]
@varyoo Ok, I've done the better thing. Not sure about the ToDnstap name though - but too tired now to think of something better.

**Comment by miekg** [comment]
@chrisohaver hats off! This is some nice work!

**Comment by miekg** [comment]
@varyoo is this OK to merge?

File: plugin/proxy/proxy.go [hunk 2]
```diff
@@ -97,7 +98,7 @@ func (p Proxy) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (
 				child.Finish()
 			}
 
-			taperr := toDnstap(ctx, host.Name, upstream.Exchanger(), state, reply, start)
+			taperr := dnstap.ToMessage(ctx, host.Name, upstream.Exchanger().Protocol(), state, reply, start)
```
Comment (2buisson) [inline plugin/proxy/proxy.go:101]: Isn't `upstream.Exchanger().Protocol()` returning "dns" or maybe "https_google"?

proxy/dns.go:
~~~ Go
func (d *dnsEx) Protocol() string { return "dns" }
~~~

This is why I introduced Exchanger.Transport() returning "tcp", "udp" or "".
"" if the request used `state.Proto()`

Comment (miekg) [reply plugin/proxy/proxy.go:101]: Fair enough, but let me rephrase: how do we want (maybe external) proxies to report dnstap? I.e. forward does not have an Exchanger?
